packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -246,18 +252,22 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange> extends
 			return;
 		}
 
+		const enrichedCommit = this.commitEnricher?.enrichCommit(commit, isResubmit) ?? commit;
 		const message = this.messageCodec.encode(
 			{
-				commit,
+				commit: enrichedCommit,
 				sessionId: this.editManager.localSessionId,
 			},
 			{
 				schema: this.schemaAndPolicy ?? undefined,
 			},
 		);
 		this.submitLocalMessage(message);
+		this.onCommitSubmitted(enrichedCommit, isResubmit);
 	}
 
+	protected onCommitSubmitted(commit: GraphCommit<TChange>, isResubmit: boolean): void {}
```

Comment:
This is only here so that the TestSharedTreeCore can sniff it, right? Instead, what do you think about making `submitCommit` protected, and have it return the enriched/submitted commit. Then TestSharedTreeCore can just override and call super, and no need for an extra concept/event.

Reply:
I love it! Thank you!